From linas@austin.ibm.com Thu Dec  1 16:59:38 2005
Date: Thu, 1 Dec 2005 18:56:14 -0600
From: <linas@austin.ibm.com>
To: greg@kroah.com
Cc: johnrose@us.ibm.com
Subject: PCI Hotplug/powerpc: remove duplicated code
Message-ID: <20051202005614.GO31651@austin.ibm.com>
Content-Disposition: inline


The RPAPHP code contains a routine that duplicates some existing code.
This patch removes the rpaphp version of the code.

Signed-off-by: Linas Vepstas <linas@austin.ibm.com>
Acked-by: John Rose <johnrose@austin.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/pci/hotplug/rpadlpar_core.c |    3 ++-
 drivers/pci/hotplug/rpaphp_pci.c    |   14 +-------------
 2 files changed, 3 insertions(+), 14 deletions(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/rpaphp_pci.c
+++ gregkh-2.6/drivers/pci/hotplug/rpaphp_pci.c
@@ -287,18 +287,6 @@ rpaphp_pci_config_slot(struct pci_bus *b
 	return dev;
 }
 
-void rpaphp_eeh_init_nodes(struct device_node *dn)
-{
-	struct device_node *sib;
-
-	for (sib = dn->child; sib; sib = sib->sibling) 
-		rpaphp_eeh_init_nodes(sib);
-	eeh_add_device_early(dn);
-	return;
-	
-}
-EXPORT_SYMBOL_GPL(rpaphp_eeh_init_nodes);
-
 static void print_slot_pci_funcs(struct pci_bus *bus)
 {
 	struct device_node *dn;
@@ -324,7 +312,7 @@ int rpaphp_config_pci_adapter(struct pci
 	if (!dn)
 		goto exit;
 
-	rpaphp_eeh_init_nodes(dn);
+	eeh_add_device_tree_early(dn);
 	dev = rpaphp_pci_config_slot(bus);
 	if (!dev) {
 		err("%s: can't find any devices.\n", __FUNCTION__);
--- gregkh-2.6.orig/drivers/pci/hotplug/rpadlpar_core.c
+++ gregkh-2.6/drivers/pci/hotplug/rpadlpar_core.c
@@ -154,7 +154,8 @@ static struct pci_dev *dlpar_pci_add_bus
 	struct pci_controller *phb = pdn->phb;
 	struct pci_dev *dev = NULL;
 
-	rpaphp_eeh_init_nodes(dn);
+	eeh_add_device_tree_early(dn);
+
 	/* Add EADS device to PHB bus, adding new entry to bus->devices */
 	dev = of_create_pci_dev(dn, phb->bus, pdn->devfn);
 	if (!dev) {
